src/scitacean/filesystem.py

```diff
@@ -86,7 +88,7 @@ def to_local(self) -> PurePath:
 
     def __truediv__(self, other: Union[str, RemotePath]) -> RemotePath:
         """Join two path segments."""
-        if isinstance(other, (PurePath, Path)):
+        if isinstance(other, (PurePath, Path)):  # type: ignore[unreachable]
             raise TypeError("OS paths are not supported when concatenating RemotePath.")
```

Comment:
I have a question how a `type-checker` would decide unreachable...
will this line below will also be considered unreachable...?
```
if not isinstance(other, (str, RemotePath)) and isinstance(other, (PurePath, Path)):
    raise ...
```

Reply:
I guess so. It must be based on the argument type hints. And mypy then assumes that no invalid type is ever passed in.